pkg/platform/abstract/platform.go

```diff
@@ -215,6 +205,50 @@ func (ap *Platform) EnrichCreateFunctionOptions(createFunctionOptions *platform.
 
 	ap.enrichMinMaxReplicas(createFunctionOptions)
 
+	// enrich with registry credential secret name
+	if createFunctionOptions.FunctionConfig.Spec.ImagePullSecrets == "" {
+		createFunctionOptions.FunctionConfig.Spec.ImagePullSecrets =
+			ap.GetDefaultRegistryCredentialsSecretName()
+	}
+
+	return nil
+}
+
+// Validate create function options against existing function
+func (ap *Platform) ValidateCreateFunctionOptionsAgainstExistingFunctionConfig(existingFunctionConfig *functionconfig.ConfigWithStatus,
+	createFunctionOptions *platform.CreateFunctionOptions) error {
+
+	// special case when we are asked to build the function and it wasn't been created yet
+	if existingFunctionConfig == nil &&
+		createFunctionOptions.FunctionConfig.Spec.Build.Mode == functionconfig.NeverBuild {
+		return errors.New("Non existing function cannot be created with neverBuild mode")
+	}
+
+	// validate resource version
+	if err := ap.ValidateResourceVersion(existingFunctionConfig, createFunctionOptions); err != nil {
+		return nuclio.WrapErrConflict(err)
+	}
+	return nil
+}
+
+// Validate existing and new create function options resource version
+func (ap *Platform) ValidateResourceVersion(existingFunctionConfig *functionconfig.ConfigWithStatus,
+	createFunctionOptions *platform.CreateFunctionOptions) error {
+
+	// existing function should always be latest
+	if existingFunctionConfig != nil {
+		existingResourceVersion := existingFunctionConfig.Meta.ResourceVersion
+		requestResourceVersion := createFunctionOptions.FunctionConfig.Meta.ResourceVersion
+
+		// when requestResourceVersion is empty, the existing one will be overridden
+		if requestResourceVersion != "" &&
+			requestResourceVersion != existingResourceVersion {
+			ap.Logger.WarnWith("Create function resource version is outdated",
+				"requestResourceVersion", requestResourceVersion,
+				"existingResourceVersion", existingResourceVersion)
+			return errors.New("Function resource version is outdated")
```

Comment:
```suggestion
			return errors.New("Function resource version is stale")
```
I believe stale is the correct term for this check. It puts emphasis on it not being "fresh"  / newest - as opposed to "outdated" which suggests that the time passed since it's creation is a factor. A `ResourceVersion` can be weeks old and still be "latest"